Tester/Framework/Assert.php

```diff
@@ -60,11 +60,11 @@ public static function same($expected, $actual)
 	 * Checks assertion. Values must not be exactly the same.
 	 * @return void
 	 */
-	public static function notSame($expected, $actual)
+	public static function notSame($expected, $actual, $message = "")
 	{
 		self::$counter++;
 		if ($actual === $expected) {
-			self::fail('%1 should not be %2', $actual, $expected);
+			self::failMessage($message, '%1 should not be %2', $actual, $expected);
 		}
```

Comment:
If we deside for example to add colors to custom message change of specialized method failMessage will be simpler I think. Because it could be something more complex than simple join in future